pkg/scalers/rabbitmq_scaler.go

```diff
@@ -229,6 +257,10 @@ func parseTrigger(meta *rabbitMQMetadata, config *ScalerConfig) (*rabbitMQMetada
 		return nil, fmt.Errorf("protocol %s not supported; must be http to use mode %s", meta.protocol, rabbitModeMessageRate)
 	}
 
+	if meta.mode == rabbitModeMessageRate && meta.useRegex {
```

Comment:
Why is this not usable with Message Rate? We currently scale (with multiple triggers for multiple queues) based on Message Rate.

Reply:
Hi,
The point here is that operations like sum are not posible with the publish rate. When you add some triggers you will have the same behavior like "max" operation, but the other operations can't be applied in a rate.
Maybe I can add another case to support also scale in base of public rate only with "max" operation but that case is already covered with the current options without this changes
Any opinion?

Reply:
Hi @rwkarg ,
any update about this topic? Can I mark as solved this conversation?

Reply:
It seems like sum and average should both work. You're looking at the rate for each queue over the same time period (what the broker is reporting) so they should sum/avg/max just fine.

Queue 1 @ 10 msg/sec
Queue 2 @ 30 msg/sec

Max would be 30 msg/sec.
Sum would be 40 msg/sec.
Avg would be 20 msg/sec.

Not sure the use case for Avg., but Sum seems to be the most useful metric to scale by. Max doesn't seem to be as useful for the cases where there could be a large number of queues.
Ex. 100 queues where one has 10/sec. and the other 99 are empty (sum is 10/sec.) 
is quite different from one queue at 10/sec and the other 99 at 9.9/sec (sum is ~1,000/sec).

Reply:
Oh... f**k me... you are right.
Maybe my brain was off the last week or idk. You are totally right, I will update my PR ASAP

Reply:
done!